File: Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/LibrarySourceProviderFactory.java
package org.cqframework.cql.cql2elm;

import java.util.Iterator;
import java.util.ServiceLoader;

public class LibrarySourceProviderFactory {
    private LibrarySourceProviderFactory() {}

    public static Iterator<LibrarySourceProvider> providers(boolean refresh) {
        var loader = ServiceLoader.load(LibrarySourceProvider.class);
        if (refresh) {
            loader.reload();
        }

        return loader.iterator();
    }
}

File: Src/java/cql-to-elm/src/main/kotlin/org/cqframework/cql/cql2elm/LibrarySourceProviderFactory.kt
package org.cqframework.cql.cql2elm

import java.util.*

object LibrarySourceProviderFactory {
    fun providers(refresh: Boolean): Iterator<LibrarySourceProvider> {
        val loader: ServiceLoader<LibrarySourceProvider> =
            ServiceLoader.load(LibrarySourceProvider::class.java)
        if (refresh) {
            loader.reload()
        }
        return loader.iterator()
    }
}
